packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationUtilities.m

```diff
@@ -0,0 +1,168 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+#import <Foundation/Foundation.h>
+#import <EXScreenOrientation/EXScreenOrientationUtilities.h>
+
+#import <sys/utsname.h>
+
+@implementation EXScreenOrientationUtilities
+
+# pragma mark - helpers
+
++ (BOOL)doesDeviceSupportOrientationMask:(UIInterfaceOrientationMask)orientationMask
+{
+  if ((UIInterfaceOrientationMaskPortraitUpsideDown & orientationMask) // UIInterfaceOrientationMaskPortraitUpsideDown is part of orientationMask
+      && [self doesDeviceHaveNotch])
+  {
+    // device does not support UIInterfaceOrientationMaskPortraitUpsideDown and it was requested via orientationMask
+    return false;
+  }
+  
+  return true;
+}
+
++ (BOOL)doesDeviceHaveNotch {
+  if (@available(iOS 11.0, *)) {
+    static BOOL result = false;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+      dispatch_sync(dispatch_get_main_queue(), ^{
+        result = ([[[UIApplication sharedApplication] delegate] window].safeAreaInsets.bottom ?: 0.0) > 0.0;
```

Comment:
Do we have a link to the source where we found this solution? Maybe we could paste it above, for future reference